packages/frontend/src/features/sqlRunner/store/barChartSlice.ts

```diff
@@ -78,14 +66,12 @@ export const barChartConfigSlice = createSlice({
             action: PayloadAction<{
                 previousReference: string;
                 reference: string;
+                index: number;
                 aggregation: AggregationOptions;
             }>,
         ) => {
             if (config?.fieldConfig?.y) {
-                const yAxis = config.fieldConfig.y.find(
-                    (axis) =>
-                        axis.reference === action.payload.previousReference,
-                );
+                const yAxis = config.fieldConfig.y[action.payload.index];
```

Comment:
fixed an issue with the y axis not being updated correctly (now that we allow multiple, it should be handled this way)

Reply:
@joaoviana you can remove the `previousReference` I think this was only being used because we weren't passing `index`

Reply:
good spot, done here: https://github.com/lightdash/lightdash/pull/10932/commits/84e5cbab2a5a6ed2a9c5522359ded287278c79b5